edgehog-device-runtime-containers/src/docker/container.rs

```diff
@@ -63,52 +66,52 @@ pub enum ContainerError {
 
 /// Docker container struct.
 #[derive(Debug, Clone, Eq)]
-pub struct Container<S>
+pub(crate) struct Container<S>
 where
     S: Hash + Eq,
 {
     /// Id of the docker container.
     ///
     /// The id of the image is optional since it will be available only when the image is created.
-    pub id: Option<String>,
+    pub(crate) id: Option<String>,
     /// Assign the specified name to the container.
     ///
     /// Must match /?[a-zA-Z0-9][a-zA-Z0-9_.-]+.
-    pub name: S,
+    pub(crate) name: S,
```

Comment:
Is it not enough to put `(crate) rule` on the struct to make visible only with crate? All the fields should be not usable since the struct is not visible outer from the crate.
``` rust
pub(crate) struct Container<S>
where
    S: Hash + Eq,
{
...    
pub id: Option<String>,
...
}
```

Reply:
Yes, that clearly works, but I think being a little bit more explicit and removing the indirection (struct private but fields public) makes it simpler to read.

I think I should have done it for all the structs/crates to begin with, but now there are a lot of them and probably a gradual change would be better.